### ISSUE TITLE
Cannot compile: No SOURCES given to target: libshared

### ISSUE THREAD

**Issue by Mebus** [GothenburgBitFactory/taskwarrior#3344]
Hallo,

after downloading the version 3.0.0 tar archive I tried to build it on Fedora 38, but I get this error message:

```
→ cmake -DCMAKE_BUILD_TYPE=release .
-- Configuring C++17
-- System: Linux
-- Looking for libshared
-- Cloning libshared
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
-- Looking for SHA1 references
-- Looking for libuuid
-- Found libuuid
-- Configuring cmake.h
-- Rust Toolchain: stable-x86_64-unknown-linux-gnu
-- Rust Target: x86_64-unknown-linux-gnu
-- Using Corrosion as a subdirectory
-- Configuring man pages
-- Configuring done (2.5s)
CMake Error at src/CMakeLists.txt:31 (add_library):
  Cannot find source file:

    libshared/src/Color.cpp


CMake Error at src/CMakeLists.txt:31 (add_library):
  No SOURCES given to target: libshared


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

What is this libshared and how can I get it?

Thanks

Kind regards

Mebus

**Comment by djmitche** [comment]
It's https://github.com/GothenburgBitFactory/libshared/, and is imported as a submodule into this repo. But, I would not expect the build process to be cloning it!

Did you download `task-3.0.0.tar.gz` or the "Source code (.tar.gz)" link? The former is the one you want. The latter is provided automatically by GitHub, and just contains a copy of the Git repo (and not libshared).

**Comment by Mebus** [comment]
Hey djmitche,

yeah, I had downloaded this tar:

https://github.com/GothenburgBitFactory/taskwarrior/archive/refs/tags/v3.0.0.tar.gz

When I use your tag v3.0.0 and do a `git submodule init` I can compile taskwarrior.

I tried this tar, too:

https://github.com/GothenburgBitFactory/taskwarrior/releases/download/v3.0.0/task-3.0.0.tar.gz

But the latter one gives me this error:

```
- Looking for libuuid
-- Looking for uuid_unparse_lower
-- Looking for uuid_unparse_lower - found
-- Found libuuid
-- Configuring cmake.h
-- Rust Toolchain: stable-x86_64-unknown-linux-gnu
-- Rust Target: x86_64-unknown-linux-gnu
-- Determining required link libraries for target x86_64-unknown-linux-gnu
-- Required static libs for target x86_64-unknown-linux-gnu: gcc_s;util;rt;pthread;m;dl;c
-- Found Rust: /home/mebus/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustc (found version "1.77.1") 
-- Using Corrosion as a subdirectory
error: failed to load manifest for workspace member `/home/mebus/programme/taskwarrior/task-3.0.0/taskchampion/integration-tests`

Caused by:
  failed to read `/home/mebus/programme/taskwarrior/task-3.0.0/taskchampion/integration-tests/Cargo.toml`

Caused by:
  No such file or directory (os error 2)
CMake Error at _deps/corrosion-src/cmake/CorrosionGenerator.cmake:24 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 101"

Call Stack (most recent call first):
  _deps/corrosion-src/cmake/CorrosionGenerator.cmake:217 (_cargo_metadata)
  _deps/corrosion-src/cmake/Corrosion.cmake:1098 (_generator_add_cargo_targets)
  src/tc/CMakeLists.txt:11 (corrosion_import_crate)
```

Mebus

**Comment by djmitche** [comment]
The latter is expected and will be fixed in the next release - see #3294. Just removing that line from Cargo.toml should fix the error for you.

Unfortunately, there's no way to disable the GitHub-provided tarballs and zip files.
